yelp_scrape.py

```diff
@@ -23,16 +24,30 @@ def url_generator(search_item, location):
     search_item = search_item.replace(' ', '+')
     location = location.replace(' ', '+').replace(',', '%2C')
     url = base_url + 'find_desc=' + search_item + '&find_loc=' + location
-    return url
-
-
-search, loc = get_search_request()
-yelp_page = requests.get(url_generator(search, loc))
-yelp_soup = BeautifulSoup(yelp_page.content, 'html.parser')
-
-test = yelp_soup.find_all('a')
-links = []
-for a in test:
-    links.append(a["href"])
-
-print(links)
+    file_name = search_item + location
+    return url, file_name
+
+# local_cache_check will take the search request url and check a local directory for a file of the same name
+# if the file is found then it will use that file instead of scraping the internet
+def local_cache_check(url, file_name):
+    # url_hash = hash(url)
+    path = input('Please enter complete filepath to search: ')
```

Comment:
Input is really easy to use and great for one-off things, but it very quickly becomes unwieldy.

I use Click (https://click.palletsprojects.com/en/8.1.x/) for command-line stuff in Python. This way, instead of inputting one argument at a time (filepath, location, etc) you can store it all in one command like

`python yelp_scrape -m -- --location "Detroit, Michigan, 48127" --filepath "xyz.html"`

then when you're testing, you can just hit the up arrow in your terminal to reuse the last command, instead of retyping all your inputs every time you want to test

saving time between being able to run code is extremely useful, people call it the "iteration speed"
the difference between having to take 30 seconds to type stuff versus being able to make a one-line change, run it, find out you had a typo, make another one-line change, and run it again all in under 2 seconds is a huge, huge difference

Reply:
So I've been playing around with click but when I use `click.option('--location', prompt='Enter location: ')` the value it returns is a function object that doesn't jive with me when I try to use that as an input in anything else. I tried the click.echo() but that doesn't seem to change anything. Is there some way to convert those objects into strings that I can't seem to find?

Reply:
I just pushed my latest in here. I wasn't sure if we would lose the comments or not. You have a lot more going on with click that I do but I am unsure what is necessary and what isn't.